rholang/src/main/scala/AbstractInterpreter/Trace.scala

```diff
@@ -0,0 +1,97 @@
+package AbstractInterpreter
+
+import cats.Monad
+
+//Trace := StateT[WriterT[List,List,(S,A)]] = S => WriterT[List,List,(S,A)]] = S => (S,A) => List[(List[S],(S,A)] = List[(A,S,List[S])]
+
+//Trace is a monad transformer that takes a state and returns a list of possible end-states,
+//a value, A, and, for each end-state, a list of intermediate states.
+
+case class Trace[S,A](reduce: S => List[(A, S, List[S])]) {
```

Comment:
Hmm, this should be a typeclass I guess?

Reply:
It doesn't look like a transformer, but a concrete monad. 
Why you don't just use `StateT[WriterT[List,List,(S,A)]]` from `cats`?

Reply:
This pull request is four days old. In the up to date version `type Trace[S,W,A] = StateT[WriterT[List,W,(S,A)]]`